packages/playwright-crawler/src/internals/adaptive-playwright-crawler.ts

```diff
@@ -175,7 +175,7 @@ export class AdaptivePlaywrightCrawler extends PlaywrightCrawler {
         override readonly config = Configuration.getGlobalConfig(),
     ) {
         super(options, config);
-        this.adaptiveRequestHandler = requestHandler;
+        this.adaptiveRequestHandler = requestHandler ?? this.router;
```

Comment:
How does it happen that `requestHandler` is `undefined`? `AdaptivePlaywrightCrawler` does not allow that. Also, `this.router` accepts the full `PlaywrightCrawlingContext`, which won't be passed to `adaptiveRequestHandler`...

Reply:
but its a good point about the context, that was the followup issue i found, since there is no `querySelector` helper then (on type level). 

and lastly, `querySelector` is not working for hidden items, tried to get `html title` and it just waits till it timeouts, and in the logs once i cancel the run i can see a lot of this

```
WARN  AdaptivePlaywrightCrawler: Reclaiming failed request back to the list or queue. locator.waitFor: Target page, context or browser has been closed
Call log:
  - waiting for locator('html title').first() to be visible
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
  -   locator resolved to hidden <title>Crawlee · Build reliable crawlers. Fast. | Crawlee</title>
```

(i wanted to showcase this on the webinar, and i was awarded with a lot of typeerrors and stuff like that, so probably better to just skip this from the live coding completely :D)

Reply:
Well, it's not optional for this one, right now, but let's make it so. Can you redeclare the implicit router in `AdaptivePlaywrightCrawler` so that it uses the correct context, too?